projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import { TimeDuration, TimeUnit, UnitStringType } from '@hypertrace/common';
+import { TableColumnConfig, TableRow } from '../../../table-api';
+import { TABLE_CELL_DATA, TABLE_COLUMN_CONFIG, TABLE_COLUMN_INDEX, TABLE_DATA_PARSER, TABLE_ROW_DATA } from '../../table-cell-injection';
+import { TableCellParserBase } from '../../table-cell-parser-base';
+import { TableCellRenderer } from '../../table-cell-renderer';
+import { TableCellRendererBase } from '../../table-cell-renderer-base';
+import { CoreTableCellParserType } from '../../types/core-table-cell-parser-type';
+import { CoreTableCellRendererType } from '../../types/core-table-cell-renderer-type';
+import { TableCellAlignmentType } from '../../types/table-cell-alignment-type';
+
+@Component({
+  selector: 'ht-duration-table-cell-renderer',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="duration-cell">
+    <span class="duration-text" [htTooltip]="this.durationText">{{ this.durationText | htDisplayString }}</span>
+    </div>
+  `,
+  styleUrls: ['./duration-table-cell-renderer.component.scss']
+})
+
+@TableCellRenderer({
+  type: CoreTableCellRendererType.Duration,
+  alignment: TableCellAlignmentType.Left,
+  parser: CoreTableCellParserType.NoOp
+})
+export class DurationTableCellRendererComponent extends TableCellRendererBase<TimeDuration> implements OnInit {
+  public durationText!: string;
+  
+  public constructor(
+    @Inject(TABLE_COLUMN_CONFIG) columnConfig: TableColumnConfig,
+    @Inject(TABLE_COLUMN_INDEX) index: number,
+    @Inject(TABLE_DATA_PARSER) parser: TableCellParserBase<TimeDuration, TimeDuration, unknown>,
+    @Inject(TABLE_CELL_DATA) cellData: TimeDuration,
+    @Inject(TABLE_ROW_DATA) rowData: TableRow
+  ) {
+    super(columnConfig, index, parser, cellData, rowData);
+  }
+
+  public ngOnInit(): void {
+    super.ngOnInit();
+
+    this.durationText = this.value.toMultiUnitString(TimeUnit.Second, true, UnitStringType.Long)
+  }
```

Comment:
you can use a method here with memoization and avoid defining constructor and init blocks.

Reply:
Better yet, just use the duration pipe and avoid both (explicit) memoization and the code. If we're looking for different behavior for the formatting, either the pipe should be updated, or we want a custom renderer. In other words, the developer expectation should be the duration cell renderer and duration pipe behave the same way.

Reply:
good point. I'd say, modify the pipe to also support long format display and use it here.

Reply:
using memoization for now, have added a todo to modify pipe (as that can also affect other places). Will modify the pipe in follow-up.

Reply:
updated pipe